docs/howto/manage-charms.md

```diff
@@ -209,23 +197,34 @@ juju deploy some-bundle --map-machines=existing,3=4,4=5
 
 ````
 
-Depending on the cloud substrate that your controller is running on, the above command will allocate a machine (physical,  virtual, LXD container) or a Kubernetes pod and then proceed to deploy the contents of the charm / bundle.
+````{dropdown} Tips for troubleshooting - machines
+Deploy on machines consists of the following steps: Provision resources/a machine M from the relevant cloud, via cloud-init maybe network config, download the `jujud` binaries from the controller, start `jujud`.
 
-```{note}
+For failure at any point, retry the `deploy` command with the `--debug` and `--verbose` flags:
 
-Depending on your use case, you may alternatively opt to provision a set of machines in advance via the `juju add-machine` command.
+```text
+juju deploy <charm> --debug --verbose
+```
 
-In this case, when running the above `juju deploy` command, Juju will detect that the model contains machines with no applications assigned to them and automatically deploy the application to one of those machines instead of spinning up a new machine.
+If it still fails,  connect to the machine and examine the logs.
```

Comment:
Do we have the cloud-init logs listed for this purpose in troubleshooting and logs? And journalctl?